### ISSUE TITLE
ICE40 LUTs with constant inputs should be assigned to 0 instead of 1

### ISSUE THREAD

**Issue by tomverbeure** [YosysHQ/yosys#514]
Here's a typical example from a Yosys generated .blif file:

```
.gate SB_LUT4 I0=$false I1=$true I2=u_blink_pll_bottom_b.count[0] I3=$false O=$techmap\u_blink_pll_bottom_b.$0\count[24:0][0]
.attr src "chip.v:48"
.param LUT_INIT 0110100110010110
```

This calculates the  LSB of an endless counter, so it's essentially an inverter and only one input of the LUT4 is assigned an active signal.

*However, I1 is assigned $true.*

When you run this code through arachne, all $true signals (except those connected to CEN) are converted to $true$2, which is a real signal that actually gets routed to all cells that have a constant input of 1. And when I check the result in the ICE40 layout viewer, I see that this consumes real routing resources.

The LOGIC Tile Documentation of Project IceStorm says the following:
> LUT inputs that are not connected to anything are driven low. The set/reset signal is also driven low if not connected to any other driver, and the clock enable signal is driven high when left unconnected.

It would be better for the LUT4 above to be output as follows:
```
.gate SB_LUT4 I0=u_blink_pll_bottom_b.count[0] I1=$false I2=$false I3=$false O=$techmap\u_blink_pll_bottom_b.$0\count[24:0][0]
.attr src "chip.v:48"
.param LUT_INIT 0000000000000001
```

In other words: assign all constant inputs to $false instead of $true.

I'm not familiar with the ICE40 enough to judge whether or not routing resources are a limiting factor on tight designs.

So is this an optimization that's worth pursuing? And, if so, is this something that would better be done inside YoSys or inside Arachne? My feeling is that the answer to these questions are 'yes' and 'Yosys'.

Tom

**Comment by tomverbeure** [comment]
<img width="871" alt="true 2 net" src="https://user-images.githubusercontent.com/36720790/37266216-00a6308a-2576-11e8-9d75-095934d8a7ea.png">
There's an example of the $true$2 net of a small design.
